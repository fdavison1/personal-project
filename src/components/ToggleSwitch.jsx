
import React from 'react'

export default function ToggleSwitch() {
    return (

        <>

        <input
            className="react-switch-checkbox"
            id={`react-switch-new`}
            type="checkbox"
        />
        <label
            className="react-switch-label"
            htmlFor={`react-switch-new`}
        >
            <span className={`react-switch-button`} />
        </label>

        </>
    )
}