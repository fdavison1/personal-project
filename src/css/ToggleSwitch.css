.react-switch-checkbox {
    height: 0;
    width: 0;
    visibility: hidden;
  }
  
  .react-switch-label {
    display: flex;
    align-items: center;
    justify-content: space-between;
    cursor: pointer;
    width: 75px;
    height: 30px;
    background: grey;
    border-radius: 100px;
    position: relative;
    transition: background-color .2s;
  }
  
  .react-switch-label .react-switch-button {
    content: '';
    position: absolute;
    top: 2px;
    left: 2px;
    width: 25px;
    height: 25px;
    border-radius: 45px;
    transition: 0.2s;
    background: #fff;
    box-shadow: 0 0 2px 0 rgba(10, 10, 10, 0.29);
  }
  
  .react-switch-checkbox:checked + .react-switch-label .react-switch-button {
    left: calc(100% - 2px);
    transform: translateX(-100%);
  }
  
  .react-switch-label:active .react-switch-button {
    width: 40px;
  }

  /* TOGGLESWITCH 2 */
  
  .onoffswitch {
    position: relative; width: 90px;
    -webkit-user-select:none; -moz-user-select:none; -ms-user-select: none;
}
.onoffswitch-checkbox {
    display: none;
}
.onoffswitch-label {
    display: block; overflow: hidden; cursor: pointer;
    border: 2px solid #999999; border-radius: 20px;
}
.onoffswitch-inner {
    display: block; width: 200%; margin-left: -100%;
    transition: margin 0.3s ease-in 0s;
}
.onoffswitch-inner:before, .onoffswitch-inner:after {
    display: block; float: left; width: 50%; height: 30px; padding: 0; line-height: 30px;
    font-size: 14px; color: white; 
    font-family: Trebuchet, Arial, sans-serif; 
    font-weight: bold;
    box-sizing: border-box;
}
.onoffswitch-inner:before {
    content: "ON";
    padding-left: 10px;
    background-color: #34A7C1; color: #FFFFFF;
}
.onoffswitch-inner:after {
    content: "OFF";
    padding-right: 10px;
    background-color: #EEEEEE; color: #999999;
    text-align: right;
}
.onoffswitch-switch {
    display: block; width: 18px; margin: 6px;
    background: #FFFFFF;
    position: absolute; top: 0; bottom: 0;
    right: 56px;
    border: 2px solid #999999; border-radius: 20px;
    transition: all 0.3s ease-in 0s; 
}
.onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-inner {
    margin-left: 0;
}
.onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-switch {
    right: 0px; 
}