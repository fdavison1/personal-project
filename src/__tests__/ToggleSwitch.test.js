import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ToggleSwitch from '../components/ToggleSwitch'

describe('ToggleSwitch.jsx', () => {
    it('renders', () => {
        const wrapper = shallow(<ToggleSwitch />)

        expect(wrapper.exists()).toBe(true)
    })
    it('matches snapshot', () => {
        const tree = renderer.create(<ToggleSwitch />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})