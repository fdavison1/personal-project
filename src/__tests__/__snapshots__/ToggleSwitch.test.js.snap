// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch.jsx matches snapshot 1`] = `
<div
  className="sc-bdVaJa gByhnJ"
>
  <div
    className="onoffswitch"
  >
    <input
      className="onoffswitch onoffswitch-checkbox"
      id="myonoffswitch"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="onoffswitch-label"
      htmlFor="myonoffswitch"
    >
      <span
        className="onoffswitch-inner"
      />
      <span
        className="onoffswitch-switch"
      />
    </label>
  </div>
</div>
`;
